src/main/java/uk/gov/hmcts/ccd/domain/service/processor/GlobalSearchProcessorService.java

```diff
@@ -0,0 +1,173 @@
+                                                          Map<String, JsonNode> data) {
+
+        Map<String, JsonNode> clonedData = null;
+        if (data != null) {
+            clonedData = new HashMap<>(data);
+        }
+
+        if (caseTypeDefinition.getCaseField(SEARCH_CRITERIA).isPresent()) {
+            List<uk.gov.hmcts.ccd.domain.model.definition.SearchCriteria> searchCriterias =
+                caseTypeDefinition.getSearchCriterias();
+
+            List<uk.gov.hmcts.ccd.domain.model.definition.SearchParty> searchParties =
+                caseTypeDefinition.getSearchParties();
+
+            SearchCriteria searchCriteria = populateSearchCriteria(clonedData, searchCriterias);
+            List<SearchParty> searchPartyList = populateSearchParties(clonedData, searchParties);
+
+            if (!searchPartyList.isEmpty()) {
+                searchCriteria.setSearchParties(searchPartyList);
+            }
+            if (!searchCriteria.isEmpty() && clonedData != null) {
+                clonedData.put(SEARCH_CRITERIA, JacksonUtils.convertValueJsonNode(searchCriteria));
+            }
+        }
+
+        return clonedData;
+    }
+
+    private SearchCriteria populateSearchCriteria(Map<String, JsonNode> data,
+                                       List<uk.gov.hmcts.ccd.domain.model.definition.SearchCriteria> searchCriterias) {
+        List<OtherCaseReference> otherCaseReferences = new ArrayList<>();
+        SearchCriteria returnValue = new SearchCriteria();
+
+        if (data != null) {
+            data.forEach((key, jsonNode) -> searchCriterias.forEach(currentSearchCriteria -> {
+
+                if (currentSearchCriteria.getOtherCaseReference().contains(".")
+                    && key.equals(currentSearchCriteria.getOtherCaseReference().split("\\.")[0])) {
+                    otherCaseReferences.add(OtherCaseReference.builder()
+                        .id(UUID.randomUUID().toString())
+                        .value(getNestedValue(jsonNode, currentSearchCriteria.getOtherCaseReference()))
+                        .build());
+                } else if (key.equals(currentSearchCriteria.getOtherCaseReference())) {
+                    otherCaseReferences.add(OtherCaseReference.builder()
+                        .id(UUID.randomUUID().toString())
+                        .value(jsonNode.textValue())
+                        .build());
+                }
+            }));
+
+            returnValue = SearchCriteria.builder().otherCaseReferences(otherCaseReferences).build();
+        }
+
+        return returnValue;
+    }
+
+    private List<SearchParty> populateSearchParties(Map<String, JsonNode> data,
+                                             List<uk.gov.hmcts.ccd.domain.model.definition.SearchParty> searchParties) {
+        List<SearchParty> searchPartyList = new ArrayList<>();
+
+        Map<String, String> namesToValuesMap = new LinkedHashMap<>();
```

Comment:
seems like this can go inside populateSearchParty(...) and no need of clear function call

Reply:
Fixed in 9f001db